### PR TITLE
Fix Prefast C6101 on MRTCore Solution

### PR DESCRIPTION
**Pull request by kythant**
A microsoft employee must use /azp run to validate using the pipelines below.

WARNING:
Comments made by azure-pipelines bot maybe inaccurate.
Please see pipeline link to verify that the build is being ran.

For status checks on the main branch, please use TransportPackage-Foundation-PR
(https://microsoft.visualstudio.com/ProjectReunion/_build?definitionId=81063&_a=summary)
and run the build against your PR branch with the default parameters.

### CODE REVIEW

File: dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ class FileBuilder : public DefObject
     UINT32 GetDataSize() { return m_cbData; }
 
 public:
-    static HRESULT CreateInstance(DEFFILE_MAGIC magic, BaseFile::SectionCount sizeSections, _Outptr_ FileBuilder** result);
+    static HRESULT CreateInstance(__in DEFFILE_MAGIC magic, __in BaseFile::SectionCount sizeSections, _Outptr_ FileBuilder** result);
```
Comment (huichen123) [inline dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h:123]: >__in [](http://example.com/codeflow?start=34&length=4)

in should not be necessary

File: dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h [hunk 2]
```diff
@@ -130,7 +130,7 @@ class FileBuilder : public DefObject
 
     HRESULT SetDescriptorIndex(__in BaseFile::SectionIndex index);
 
-    HRESULT GetSectionData(INT32 sectionIndex, _Out_ const BYTE** data, _Out_ UINT32* pcbSectionData);
+    HRESULT GetSectionData(__in INT32 sectionIndex, _Out_ const BYTE** data, __out UINT32* pcbSectionData);
```
Comment (huichen123) [inline dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h:133]: >__out [](http://example.com/codeflow?start=77&length=5)

why mix SAL1 and SAL2?

File: dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h (hunk 1 above)
Comment (kythant) [reply dev/MRTCore/mrt/mrm/include/mrm/build/FileBuilder.h:123]: In FileBuilder.cpp, it defines

HRESULT FileBuilder::CreateInstance(__in DEFFILE_MAGIC magic, __in BaseFile::SectionCount sizeSections, _Outptr_ FileBuilder** result)

**Review by huichen123 — APPROVED** [review]
:shipit:

**Pull request merged**